modules/core/src/main/java/org/apache/ignite/internal/processors/cache/transactions/IgniteTxLocalAdapter.java

```diff
@@ -937,7 +938,15 @@ assert ownsLock(txEntry.cached()) :
                     U.warn(log, "Failed to commit transaction, node is stopping [tx=" + CU.txString(this) +
                         ", err=" + ex + ']');
 
-                    return;
+                    boolean persistenceEnabled = CU.isPersistenceEnabled(cctx.kernalContext().config());
+
+                    if (persistenceEnabled) {
+                        GridCacheDatabaseSharedManager dbManager = (GridCacheDatabaseSharedManager)cctx.database();
+
+                        dbManager.skipCheckPointOnNodeStop(true);
```

Comment:
Why not to replace this call with something like `dbManager.getCheckpointer().skipCheckpointOnNodeStop(true)`?

By this we'll eliminate two unnecessary methods on dbManager and checkpointManager.

What do you think?